db/collection.go

```diff
@@ -451,20 +458,17 @@ func (t *Txn) Discard() {
 	t.discarded = true
 }
 
-func getInstanceID(t []byte) core.InstanceID {
+func getInstanceID(t []byte) (id core.InstanceID, err error) {
 	partial := &struct {
 		ID *string `json:"_id"`
 	}{}
 	if err := json.Unmarshal(t, partial); err != nil {
-		log.Fatalf("error when unmarshaling json instance: %v", err)
+		return id, fmt.Errorf("error unmarshaling json instance: %v", err)
 	}
 	if partial.ID == nil {
-		log.Fatal("invalid instance: doesn't have an _id attribute")
-	}
-	if *partial.ID != "" && !core.IsValidInstanceID(*partial.ID) {
-		log.Fatal("invalid instance: invalid _id value")
+		return id, errMissingInstanceID
```

Comment:
Think we should return `core.EmptyInstanceID` explicitly here since that's what is tested for to decide if we need to generate an id for the new instance?

Reply:
Here and a couple lines above as well

Reply:
Ah nice, good call. Just pushed the change.